.github/workflows/test-pg_search.yml

```diff
@@ -121,111 +124,135 @@ jobs:
           sudo PG_CONFIG=/usr/lib/postgresql/${{ matrix.pg_version }}/bin/pg_config make -j
           sudo PG_CONFIG=/usr/lib/postgresql/${{ matrix.pg_version }}/bin/pg_config make install -j
 
-      - name: Extract pgrx version
+      - name: Install llvm-tools-preview
         if: steps.check_skip.outputs.skip_remaining_steps != 'true'
-        working-directory: pg_search/
-        run: echo "PGRX_VERSION=$(cargo tree --depth 1 -i pgrx -p pg_search | head -n 1 | cut -f2 -dv)" >> $GITHUB_ENV
+        run: rustup component add llvm-tools-preview
 
-      - name: Install pgrx & llvm-tools-preview
+      - name: Install cargo-pgrx
         if: steps.check_skip.outputs.skip_remaining_steps != 'true'
-        run: |
-          cargo install -j $(nproc) --locked cargo-pgrx --version ${{ env.PGRX_VERSION }} --debug
-          rustup component add llvm-tools-preview
-          cargo pgrx init "--pg${{ matrix.pg_version }}=/usr/lib/postgresql/${{ matrix.pg_version }}/bin/pg_config"
+        run: cargo install -j $(nproc) --locked cargo-pgrx --version ${{ env.PGRX_VERSION }} --debug
+
+      - name: Initialize cargo-pgrx environment
+        if: steps.check_skip.outputs.skip_remaining_steps != 'true'
+        run: cargo pgrx init "--pg${{ matrix.pg_version }}=/usr/lib/postgresql/${{ matrix.pg_version }}/bin/pg_config"
 
       - name: Add pg_search to shared_preload_libraries
         if: steps.check_skip.outputs.skip_remaining_steps != 'true'
         working-directory: /home/runner/.pgrx/data-${{ matrix.pg_version }}/
         run: sed -i "s/^#shared_preload_libraries = .*/shared_preload_libraries = 'pg_search'/" postgresql.conf
 
-      - name: Run pg_search Cargo Test Suite
+      - name: Install pg_search extension
+        if: steps.check_skip.outputs.skip_remaining_steps != 'true'
+        working-directory: pg_search/
+        run: cargo pgrx install --features icu --pg-config="/usr/lib/postgresql/${{ matrix.pg_version }}/bin/pg_config"
+
+      - name: Start Postgres via cargo-pgrx
         if: steps.check_skip.outputs.skip_remaining_steps != 'true'
         working-directory: pg_search/
         run: |
-          # # Variables (we disable telemetry to avoid skewing the user metrics with CI runs)
```

Comment:
Don't we want to reactivate this instead to test extension upgrade?

Reply:
you'll notice I added a comment that it should be turned into a shell script instead of inline in the .yml.

also, we don't need to do that on every PR.  We should probably do it either on a schedule or sometime late in our release process.

Reply:
It only runs on PRs to `main` as is currently built

Reply:
it really ought to be a separate job of some kind.  I feel like that's a step too far to have all jumbled up in a single CI yaml file.

Reply:
Fair enough, happy to move it

Reply:
> Fair enough, happy to move it

I did.  :)  We'll make some kind of script that can test upgrades from any version to any other version and figure out how to get that into CI at the right phase of our processes.